nameable.rb

```diff
@@ -0,0 +1,30 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
+  end
+end
```

Comment:
- [ ] [optional] @MasumaJaffery great work implementing all the changes requested, it would have been better if each class was in its own file, this will make it easier to read and scale your project in the future.